diff --git a/drivers/staging/imgtec/rogue/rk3368/sysconfig.c b/drivers/staging/imgtec/rogue/rk3368/sysconfig.c
index b984f98..e655615 100644
--- a/drivers/staging/imgtec/rogue/rk3368/sysconfig.c
+++ b/drivers/staging/imgtec/rogue/rk3368/sysconfig.c
@@ -150,7 +150,7 @@ PVRSRV_ERROR SysDevInit(void *pvOSDevice, PVRSRV_DEVICE_CONFIG **ppsDevConfig)
 	 * Setup RGX specific timing data
 	 */
 	gsRGXTimingInfo.ui32CoreClockSpeed        = RGX_RK_CORE_CLOCK_SPEED;
-	gsRGXTimingInfo.bEnableActivePM           = IMG_TRUE;
+	gsRGXTimingInfo.bEnableActivePM           = IMG_FALSE;
 	gsRGXTimingInfo.bEnableRDPowIsland        = IMG_FALSE;
 	gsRGXTimingInfo.ui32ActivePMLatencyms     = SYS_RGX_ACTIVE_POWER_LATENCY_MS;
 
